pkg/logs/input/docker/decoder.go

```diff
@@ -0,0 +1,74 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+// +build docker
+
+package docker
+
+import (
+	"bytes"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/decoder"
+)
+
+// Decoder specific to docker container source.
+type Decoder struct {
+	*decoder.Decoder
+}
+
+// InitializeDecoder returns a properly initialized Decoder
+func InitializeDecoder(source *config.LogSource, containerID string) *Decoder {
+
+	return &Decoder{
+		Decoder: decoder.NewDecoderWithStrategy(source, NewParser(containerID), &matchHeader{}),
+	}
+}
+
+// Start starts the Decoder
+func (d *Decoder) Start() {
+	d.Decoder.Start()
+}
+
+// Stop stops the Decoder
+func (d *Decoder) Stop() {
+	d.Decoder.Stop()
+}
+
+const (
+	dockerHeaderLen       = 8
```

Comment:
as you're already in the docker package, you can remove all docker prefixes in this file.